game.sy

```diff
@@ -99,8 +99,75 @@ player_new :: fn pos: (float, float) -> Player {
 
 :Player is :Entity
 
+Puck :: blob {
+    pos: (float, float),
+    rot: float,
+    vel: float,
+    acc: float,
+    holder: Player | void,
+    hold_cooldown: float,
+    draw: fn -> void,
+    process_input: fn Input, float -> void,
+    update: fn float -> void,
+}
+
+puck_new :: fn pos: (float, float) -> Puck {
+    self: !Puck = nil
+    self = Puck{
```

Comment:
Not really related to this PR but is this something we want to be able to do without the nil? @FredTheDino

Reply:
Yes - I want to add a special case in the compiler for it.